app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+class PostsController < ApplicationController
+  def new
```

Comment:
- [ ] `PostsController` uses `before_action` to restrict access to the :new and :create methods (uses ApplicationController#signed_in?)
```ruby
before_action :signed_in_only!, only: [:new, :create]

def signed_in_only!
  # signed_in? is defined in ApplicationController
  redirect_to sign_in_path unless signed_in?
end
```